pkgs/development/tools/k6/default.nix

```
@@ -1,26 +1,34 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
 
 buildGoModule rec {
   pname = "k6";
-  version = "0.38.3";
+  version = "0.39.0";
 
   src = fetchFromGitHub {
     owner = "grafana";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-MV5GbsXVvq99tI5LCK6VgcXRtNUfffoz3FopwPljhdA=";
+    sha256 = "sha256-fphhXbaK5wNhBaP8+d4Ktqf4G8OyX/1SLiHVF+jlUF0=";
   };
 
   subPackages = [ "./" ];
 
   vendorSha256 = null;
 
-  doCheck = true;
+  nativeBuildInputs = [ installShellFiles ];
+
   doInstallCheck = true;
   installCheckPhase = ''
     $out/bin/k6 version | grep ${version} > /dev/null
   '';
 
+  postInstall = ''
```

To not break cross compilation:

```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
```

---

Welp, can you explain/reference a bit this part?

I have a postInstall to install shell files in a bunch of other Golang based projects.

---

I guess it is because you are using `$out/bin/k6` in the hook.

---

Yes. If you cross compile i.e. hostPlatform is aarch64 and buildPlatform is x86_64, than you cannot run your $out/bin/k6 binary compiled for aarch64 on x86_64.

---

Thanks @smancill @Mic92, will keep in mind to refactor my packages.